packages/meditrak-server/src/apiV2/import/importSurveyResponses/importSurveyResponses.js

```diff
@@ -103,25 +99,38 @@ export async function importSurveyResponses(req, res) {
       throw new UploadError();
     }
     const { models, query, userId } = req;
-    const surveyNames = getArrayQueryParameter(query.surveyNames);
+
     const { timeZone } = query;
     if (!timeZone) {
       throw new Error(`Timezone is required`);
     }
     const updatePersistor = new SurveyResponseUpdatePersistor(models);
+
     const workbook = xlsx.readFile(req.file.path);
-    // Go through each sheet in the workbook and process the updated survey responses
-    const entitiesBySurveyName = await getEntitiesBySurveyName(
+    const tabNames = Object.keys(workbook.Sheets);
+    const querySurveyCodes = getArrayQueryParameter(query.surveyCodes);
+
+    if (querySurveyCodes) {
+      for (const querySurveyCode of querySurveyCodes) {
+        if (!tabNames.includes(querySurveyCode)) {
+          throw new ImportValidationError(
+            `Survey with code ${querySurveyCode} specified in import but there is no tab named ${querySurveyCode} in the spreadsheet.`,
+          );
+        }
+      }
+    }
+
+    const entitiesBySurveyCode = await getEntitiesBySurveyCode(
       models,
       workbook.Sheets,
-      surveyNames,
+      querySurveyCodes,
```

Comment:
I would have thought this needs to take `tabNames` if no survey codes are passed in the query?

Reply:
Ah this argument was obsolete after some cleanup, removed.